### PR TITLE
chore: Add SPDX headers and REUSE CI workflow

### PR DESCRIPTION
**Pull request by susnux**
<del>* :warning: this is chained on #5647</del>

### ☑️ Resolves

Add SPDX headers to all files 

### 🏁 Checklist

- [x] ⛑️ Tests are included or are not applicable
- [x] 📘 Component documentation has been extended, updated or is not applicable
- [ ] 3️⃣ Backport to `next` requested with a Vue 3 upgrade

### CODE REVIEW

File: .reuse/dep5 [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: Nextcloud Vue components library
+Upstream-Contact: Nextcloud GmbH <https://nextcloud.com/impressum/>
+Source: https://github.com/nextcloud-libraries/nextcloud-vue
+
+Files: package.json package-lock.json .github/pull_request_template.md
+Copyright: Nextcloud GmbH and Nextcloud contributors
```
Comment (AndyScherzinger) [inline .reuse/dep5:7]: <year> if available (optional but recommended)

File: .reuse/dep5 [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+License: AGPL-3.0-or-later
+
+Files: styleguide/assets/* cypress/snapshots/* tests/unit/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap
+Copyright: Nextcloud GmbH and Nextcloud contributors
```
Comment (AndyScherzinger) [inline .reuse/dep5:11]: <year> if available (optional but recommended)

**Comment by AndyScherzinger** [comment]
The year isn't mandatory but strongly suggested I would say.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,10 @@
 # Vue components
 
+<!--
+  - SPDX-FileCopyrightText: Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+
```
Comment (ShGKme) [inline README.md:7]: Shouldn't it be in the top of the file? And same in other MD files

File: .reuse/dep5 (hunk 2 above)
Comment (ShGKme) [inline .reuse/dep5:11]: I'm not sure all the files in this folder have this license. app-background, for example, or twitter logo.

Comment (susnux) [reply .reuse/dep5]: Adjusted most of them, removed the unused other

Comment (susnux) [reply .reuse/dep5:11]: Added :)

**Review by AndyScherzinger — APPROVED** [review]
Fine by me to merge, still some files missing the year info, which is fine I guess, any header is better than none 👍

File: .reuse/dep5 [hunk 4]
```diff
@@ -0,0 +1,40 @@
+Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: Nextcloud Vue components library
+Upstream-Contact: Nextcloud GmbH <https://nextcloud.com/impressum/>
+Source: https://github.com/nextcloud-libraries/nextcloud-vue
+
+Files: package.json package-lock.json .github/pull_request_template.md
+Copyright: Nextcloud GmbH and Nextcloud contributors
+License: AGPL-3.0-or-later
+
+Files: styleguide/assets/default.css styleguide/assets/icons.css styleguide/assets/server.css cypress/snapshots/* tests/unit/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap
```
Comment (ShGKme) [inline .reuse/dep5:10]:
```suggestion
Files: styleguide/assets/icons.css styleguide/assets/server.css cypress/snapshots/* tests/unit/functions/usernameToColor/__snapshots__/usernameToColor.spec.js.snap
```

This files already has SPDX comment

File: .reuse/dep5 [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+Copyright: Internet Assigned Numbers Authority
+License: CC0-1.0
+
+Files: styleguide/assets/img/favicon*.* styleguide/assets/img/logo/logo*.*
+Copyright: 2016-2024 Nextcloud GmbH
+License: LicenseRef-NextcloudTrademarks
+
+Files: styleguide/assets/close.svg styleguide/assets/img/filetypes/*
+Copyright: 2020 Google Inc.
+License: Apache-2.0
```
Comment (ShGKme) [inline .reuse/dep5:32]: Are all the filetypes images Google images?
Also, could you add extensions here to the glob to have less risk of having wrong files?

And what about other images?

File: .reuse/dep5 (hunk 4 above)
Comment (susnux) [reply .reuse/dep5:10]: That's new

File: .reuse/dep5 (hunk 5 above)
Comment (susnux) [reply .reuse/dep5:32]: All mdi now

File: .reuse/dep5 (hunk 4 above)
Comment (susnux) [reply .reuse/dep5]: Resolved

File: .reuse/dep5 (hunk 5 above)
Comment (susnux) [reply .reuse/dep5]: Resolved

**Review by ShGKme — APPROVED** [review]
There are still globs in reuse with glob extension (see comment above). I was told by @nickvergessen that this is not good as may potentially take more files than needed.

Other than that looks fine. Checked with https://github.com/nextcloud/github_helper/tree/master/spdx-convertor adjusted to this repo.

**Comment by AndyScherzinger** [comment]
@ShGKme you mean https://github.com/nextcloud-libraries/nextcloud-vue/pull/5650#discussion_r1626513344 as in the wildcards are to wide?

**Comment by AndyScherzinger** [comment]
@susnux I'll take care of fixing the images and licenses in reuse 👍

**Comment by ShGKme** [comment]
> @ShGKme you mean [#5650 (comment)](https://github.com/nextcloud-libraries/nextcloud-vue/pull/5650#discussion_r1626513344) as in the wildcards are to wide?

Yes, but I haven't found Joas's comment about it in Talk Desktop's PR, may be It was in my dream.

*confirming with Joas*

**Review by ShGKme — CHANGES_REQUESTED** [review]
No, this is not correct with this glob, same as it was on server.
Some files are not from Nextcloud, see:
https://github.com/nextcloud/server/blob/59c181b4cbc47821d0a7d35df747a8b1d012dfc7/.reuse/dep5#L158-L160

We should copy this block from server (and somehow maintain it 🥲)

**Comment by AndyScherzinger** [comment]
> No, this is not correct with this glob, same as it was on server.

Ah, yes. That I will fix  myself on this PR since I also did that on server including replacing the icon for Material Symbol ones 👍 

Thanks for the feedback @ShGKme - now it is clear to me what needs to be done 👍

**Comment by AndyScherzinger** [comment]
@ShGKme see https://github.com/nextcloud-libraries/nextcloud-vue/pull/5650/commits/b56a8b5b0bbfa94bfa490892fbf30fd20ccc06d8 for replaced icons and license info 👍

**Review by ShGKme — APPROVED** [review]

**Comment by AndyScherzinger** [comment]
@susnux could you maybe take care of the forward porting to `next`? No hurries/rush

**Comment by susnux** [comment]
@AndyScherzinger yes

**Pull request merged**